File: NutriSmart/app/src/main/java/com/example/nutrismart/data/repository/UserRepository.kt
package com.example.nutrismart.data.repository

import com.example.nutrismart.data.model.UpdateUserRequest
import com.example.nutrismart.data.model.UserDto
import com.example.nutrismart.data.remote.NutriSmartApi
import retrofit2.Response
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class UserRepository @Inject constructor(
    private val api: NutriSmartApi
) {
    suspend fun getUser(token: String, userId: Long): Response<UserDto> {
        return api.getUser("Bearer $token", userId)
    }

    suspend fun patchUser(token: String, userId: Long, request: UpdateUserRequest): Response<UserDto> {
        return api.patchUser("Bearer $token", userId, request)
    }

    suspend fun deleteUser(token: String, userId: Long): Response<Void> {
        return api.deleteUser("Bearer $token", userId)
    }

    suspend fun completeProfile(token: String, request: com.example.nutrismart.data.model.OnboardingRequest): Response<Any> {
        return api.completeProfile("Bearer $token", request)
    }

    suspend fun startPlanGeneration(token: String, userId: Long): Response<Map<String, String>> {
        return api.startPlanGeneration("Bearer $token", userId)
    }

    suspend fun checkGenerationStatus(token: String, userId: Long): Response<Map<String, String>> {
        return api.checkGenerationStatus("Bearer $token", userId)
    }

    suspend fun searchFoods(token: String, query: String): Response<List<String>> {
        return api.searchFoods("Bearer $token", query)
    }
}

File: NutriSmartIOS/data/repository/UserRepository.swift
//
//  UserRepository.swift
//  NutriSmartIOS
//

import Foundation

class UserRepository {
    static let shared = UserRepository()
    
    private let api = NutriSmartApiService.shared
    
    private init() {}
    
    func getUser(token: String, userId: Int) async throws -> UserDto {
        return try await api.getUser(token: token, userId: userId)
    }
    
    func patchUser(token: String, userId: Int, request: UpdateUserRequest) async throws -> UserDto {
        return try await api.patchUser(token: token, userId: userId, request: request)
    }
    
    func deleteUser(token: String, userId: Int) async throws -> EmptyResponse {
        return try await api.deleteUser(token: token, userId: userId)
    }
    
    func completeProfile(token: String, request: OnboardingRequest) async throws -> EmptyResponse {
        return try await api.completeProfile(token: token, request: request)
    }
    
    func startPlanGeneration(token: String, userId: Int) async throws -> [String: String] {
        return try await api.startPlanGeneration(token: token, userId: userId)
    }
    
    func checkGenerationStatus(token: String, userId: Int) async throws -> [String: String] {
        return try await api.checkGenerationStatus(token: token, userId: userId)
    }
    
    func searchFoods(token: String, query: String) async throws -> [String] {
        return try await api.searchFoods(token: token, query: query)
    }
}
